src/testing/burgers_energy_conservation_rrk.cpp

```diff
@@ -0,0 +1,161 @@
+#include "physics/exact_solutions/exact_solution.h"
+#include "cmath"
+
+namespace PHiLiP {
+namespace Tests {
+
+template <int dim, int nstate>
+BurgersEnergyConservationRRK<dim, nstate>::BurgersEnergyConservationRRK(
+        const PHiLiP::Parameters::AllParameters *const parameters_input,
+        const dealii::ParameterHandler &parameter_handler_input)  
+        : TestsBase::TestsBase(parameters_input),
+         parameter_handler(parameter_handler_input)
+{}
+
+template <int dim, int nstate>
+Parameters::AllParameters BurgersEnergyConservationRRK<dim,nstate>::reinit_params(bool use_rrk, double time_step_size) const
+{
+    PHiLiP::Parameters::AllParameters parameters = *(this->all_parameters);
+    
+    parameters.ode_solver_param.initial_time_step = time_step_size;
+    
+    using ODESolverEnum = Parameters::ODESolverParam::ODESolverEnum;
+    if (use_rrk)    {parameters.ode_solver_param.ode_solver_type = ODESolverEnum::rrk_explicit_solver;}
+    else            {parameters.ode_solver_param.ode_solver_type = ODESolverEnum::explicit_solver;}
+
+    return parameters;
+}
+
+template <int dim, int nstate>
+int BurgersEnergyConservationRRK<dim, nstate>::compare_energy_to_initial(
+        const std::shared_ptr <DGBase<dim, double>> dg,
+        const double initial_energy,
+        bool expect_conservation
+        ) const{
+    
+    const double final_energy = compute_energy_collocated(dg);
+    const double energy_change = abs(initial_energy-final_energy);
+    pcout << "Energy change at end was " << energy_change <<std::endl;
+    if (expect_conservation && (energy_change< 1E-13)){
+        pcout << "Energy was conserved, as expected." << std::endl;
+        return 0; //pass test
+    } else if (!expect_conservation && (energy_change > 1E-13)){
+        pcout << "Energy was NOT conserved, as expected." << std::endl;
+        return 0; //pass test
+    }else if (expect_conservation && (energy_change > 1E-13)){
+        pcout << "Energy was NOT conserved, but was expected to be conserved." << std::endl;
+        pcout << "    Unexpected result! Test failing." << std::endl;
+        return 1; //fail test
+    }else{
+        pcout << "Energy was conserved, but was expected NOT to be conserved." << std::endl;
+        pcout << "    Unexpected result! Test failing." << std::endl;
+        return 1; //fail test (included for completeness, but not expected to be used)
+    }
+}
+
+template <int dim, int nstate>
+double BurgersEnergyConservationRRK<dim, nstate>::compute_energy_collocated(
+        const std::shared_ptr <DGBase<dim, double>> dg
+        ) const
+{
+    double energy = 0.0;
+    for (unsigned int i = 0; i < dg->solution.size(); ++i)
+    {
+        energy += 1./(dg->global_inverse_mass_matrix.diag_element(i)) * dg->solution(i) * dg->solution(i);
+    }
+    return energy;
+}
+
+template <int dim, int nstate>
+int BurgersEnergyConservationRRK<dim,nstate>::run_flow_solver(
```

Comment:
rename to `get_energy_and_compare_to_initial`? Since its doing more than just `flow_solver->run()`

Reply:
Yep, that's a good suggestion. Changing in the next commit.